ui/admin-portal/src/app/components/candidates/show/candidate-source-base.ts

```diff
@@ -84,6 +84,10 @@ export class CandidateSourceBaseComponent {
   }
 
   protected loadSelectedFields() {
+    // Service and child components need to know the candidate source type
+    // (e.g. list (submission/regular), search) to set available fields
+    this.candidateFieldService.setCandidateSource(this.candidateSource);
```

Comment:
I don't understand the need for this. The call to candidateFieldService passes in the candidate source.

Also - this is not a good way to pass in state to a service - which can be used from multiple places. It would fail in a multithreaded environment. That may not be a problem in Angular which operates in a single thread I believe - but it is not good practice anyway.

Reply:
Ah fair enough — it was a lot more code to pass the source without a service property, but I've read up and thought on it and fully take your point. Have pushed changes for your consideration.